extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java

```diff
@@ -15,7 +15,9 @@ public ArcThreadSetupAction(ManagedContext managedContext) {
 
     @Override
     public ThreadState activateInitial() {
-        managedContext.activate();
+        if (!managedContext.isActive()) {
```

Comment:
I think that you also need to make the returned `ThreadState` a no-op.

Reply:
I don't think so, but I'll have to look closer

Reply:
Well, if you don't activate the context then you should not destroy it either :shrug:

Reply:
Let's see what this latest change does.

I still need to add an test that uses Reactive Routes and RESTEasy Reactive